app/notifications/mentee_application/rejection_notification.rb

```diff
@@ -0,0 +1,25 @@
+# To deliver this notification:
+#
+# MenteeApplication::RejectionNotification.with(post: @post).deliver_later(current_user)
+# MenteeApplication::RejectionNotification.with(post: @post).deliver(current_user)
+
+class MenteeApplication::RejectionNotification < Noticed::Base
+  deliver_by :email,
+    mailer: 'MenteeApplicationMailer',
+    method: :notify_for_rejection,
+    enqueue: true
```

Comment:
TIL: `enqueue: true` is necessary for email notifications to be enqueued as background jobs. Without this, the tests that checked that there was a Mailer enqueued would fail. I think being explicit here is helpful when TDDing and making sure that the mailer work is performed asynchronously.